key.go

```diff
@@ -96,84 +97,90 @@ func (d *DEK) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
-// KeyIterator iterates over list of KeyDescription objects.
-//   for iterator.Next() {
-//       _ = iterator.Value() // Use the KeyDescription
-//   }
-//   if err := iterator.Err(); err != nil {
-//   }
-//   if err := iterator.Close(); err != nil {
-//   }
-//
-// Once done with iterating over the list of KeyDescription
-// objects, an iterator should be closed using the Close
-// method.
-//
-// In general, a KeyIterator does not provide any guarantees
-// about ordering or the when its underlying source is modified
-// concurrently.
-// Particularly, if a key is created or deleted at the KES server
-// the KeyIterator may or may not be affected by this change.
-type KeyIterator struct {
-	response *http.Response
-	decoder  *json.Decoder
-
-	last     KeyDescription
-	nextErr  error // error encountered in Next()
-	closeErr error // error encountered in Close()
-	closed   bool
+// KeyInfo describes a cryptographic key at a KES server.
+type KeyInfo struct {
+	Name      string    // Name of the cryptographic key
+	CreatedAt time.Time // Point in time when the key was created
+	CreatedBy Identity  // Identity that created the key
 }
 
-// KeyDescription describes a cryptographic key at a KES server.
-type KeyDescription struct {
-	// Name is the name of the cryptographic key.
-	Name string `json:"name"`
+// KeyIterator iterates over a stream of KeyInfo objects.
+// Close the KeyIterator to release associated resources.
+type KeyIterator struct {
+	decoder *json.Decoder
+	closer  io.Closer
+
+	current KeyInfo
+	err     error
+	closed  bool
 }
 
-// Next returns true if there is another KeyDescription.
-// This KeyDescription can be retrieved via the Value method.
+// Value returns the current KeyInfo. It returns
+// the same KeyInfo until Next is called again.
 //
-// It returns false once there is no more KeyDescription
-// or if the KeyIterator encountered an error. The error,
-// if any, can be retrieved via the Err method.
+// If KeyIterator has been closed or if Next has not been
+// called once resp. once Next returns false then the
+// behavior of Value is undefined.
+func (i *KeyIterator) Value() KeyInfo { return i.current }
+
+// Name returns the name of the current key. It is a
+// short-hand for Value().Name.
+func (i *KeyIterator) Name() string { return i.current.Name }
+
+// CreatedAt returns the created-at timestamp of the current
+// key. It is a short-hand for Value().CreatedAt.
+func (i *KeyIterator) CreatedAt() time.Time { return i.current.CreatedAt }
+
+// CreatedBy returns the identiy that created the current key.
+// It is a short-hand for Value().CreatedBy.
+func (i *KeyIterator) CreatedBy() Identity { return i.current.CreatedBy }
+
+// Next returns true if there is another KeyInfo.
+// It returns false if there are no more KeyInfo
+// objects or when the KeyIterator encounters an
+// error.
 func (i *KeyIterator) Next() bool {
```

Comment:
How about making it more "streaming-like"?
```suggestion
// Next returns nil if there is another KeyInfo.
// It returns io.EOF if there are no more KeyInfo
// objects.
// Any other error indicates an error was encountered,
// and there are no more entries.
func (i *KeyIterator) Next() error {
```